src/us/kbase/workspace/database/provenance/EventDate.java

```diff
@@ -0,0 +1,96 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import us.kbase.workspace.database.Util;
+
+/**
+ * The date of a specified event.
+ */
+public class EventDate {
+
+	private final String date;
+	private final String dateFormat;
+	private final String event;
+
+	public static final Pattern VALID_DATE_FORMAT_REGEX = Pattern.compile("^YYYY(-MM(-DD)?)?$");
+
+	public static final Pattern VALID_DATE_REGEX = Pattern.compile("^[12]\\d{3}(-(0[1-9]|1[0-2])(-(0[1-9]|[12]\\d|3[01]))?)?$");
+
+	private EventDate(
+		final String date,
+		final String dateFormat,
+		final String event) {
+		this.date = date;
+		this.dateFormat = dateFormat;
+		this.event = event;
+	}
+
+	/** Get the date, for example "2022-05-10".
+	 * @return the date.
+	 */
+	public String getDate() {
+		return date;
+	}
+
+	/** Get the format of the date, for example "YYYY-MM-DD".
+	 * @return the date format.
+	 */
+	public String getDateFormat() {
+		return dateFormat;
+	}
+
+	/** Get the event that occurred on the date in question, for example "updated".
+	 * @return the event.
+	 */
+	public String getEvent() {
```

Comment:
Is there any controlled vocab here?

Reply:
Potentially, although I need to discuss this further with Elisha _et al._  at an upcoming CE meeting. For now I will leave as-is.

Reply:
:+1: